azure-kusto-ingest/source/ingestionProperties.ts

```diff
@@ -52,28 +61,211 @@ export enum ReportMethod {
+
+        if (this.Properties) {
+            result.Properties = {};
+            for (const key in this.Properties) {
+                if (this.Properties.hasOwnProperty(key)) {
+                    const typedKey = key as keyof MappingProperties;
+                    const property = this.Properties[typedKey];
+                    if (property !== undefined && property !== null) {
+                        result.Properties[typedKey] = property.toString();
+                    }
+                }
+            }
+        }
+        return result;
+    }
 }
 
 export class CsvColumnMapping extends ColumnMapping {
-    constructor(readonly columnName: string, readonly cslDataType: string, readonly ordinal: string) {
-        super();
+    /**
+     * @deprecated Use the factory methods instead.
+     */
+    protected constructor(readonly columnName: string, readonly cslDataType?: string, readonly ordinal?: string, constantValue?: string) {
+        super(columnName, cslDataType, { Ordinal: ordinal === undefined ? undefined : parseInt(ordinal, 10), ConstValue: constantValue });
+    }
+
+    public static withOrdinal(columnName: string, ordinal: number, cslDataType?: string): CsvColumnMapping {
+        return new CsvColumnMapping(columnName, cslDataType, ordinal.toString());
+    }
+
+    public static withConstantValue(columnName: string, constantValue: string, cslDataType?: string): CsvColumnMapping {
+        return new CsvColumnMapping(columnName, cslDataType, undefined, constantValue);
     }
+
+    mappingType = (): IngestionMappingType => IngestionMappingType.CSV;
 }
 
 export class JsonColumnMapping extends ColumnMapping {
-    constructor(readonly columnName: string, readonly jsonPath: string, readonly cslDataType: string | null = null) {
-        super();
+    /**
+     * @deprecated Use the factory methods instead.
+     */
+    constructor(readonly columnName: string, readonly jsonPath?: string, cslDataType: string | null = null, constantValue?: string, transform?: Transformation) {
+        super(columnName, cslDataType ?? undefined, { Path: jsonPath, ConstValue: constantValue, Transform: transform });
+    }
+
+    public static withPath(columnName: string, path: string, cslDataType?: string, transform?: FieldTransformation): JsonColumnMapping {
+        return new JsonColumnMapping(columnName, path, cslDataType, undefined, transform);
+    }
+
+    public static withConstantValue(columnName: string, constantValue: string, cslDataType?: string): JsonColumnMapping {
+        return new JsonColumnMapping(columnName, undefined, cslDataType, constantValue);
+    }
+
+    public static withTransform(columnName: string, transform: ConstantTransformation, cslDataType?: string): JsonColumnMapping {
+        return new JsonColumnMapping(columnName, undefined, cslDataType, undefined, transform);
     }
+
+    mappingType = (): IngestionMappingType => IngestionMappingType.JSON;
 }
 
+export class AvroColumnMapping extends ColumnMapping {
```

Comment:
Please remove all these
JsonMapping is enough

Reply:
Not even CsvMapping?

Reply:
Discussed with team - keeping this approach